### PR TITLE
build: update extern tags

### PR DESCRIPTION
**Pull request by fredevb**
- Fixed some implicit conversion warnings when enabling -Wfloat-conversion.
- Updated detray tag to v0.72.1
- Updated covfie tag to v0.10.0

These changes are needed for the traccc plugin to pass the ACTS CI.

### CODE REVIEW

**Comment by stephenswat** [comment]
If it helps, there are tools that you can use to run GitHub Actions-like CI jobs on your local machine so you don't need to constantly wait for the CI to finish. Check out https://github.com/nektos/act.

**Comment by stephenswat** [comment]
Hi @fredevb, what's the status on this; do you need any help getting this sorted out?

File: cmake/traccc-config.cmake.in [hunk 1]
```diff
@@ -40,7 +40,9 @@ endif()
 if( TRACCC_BUILD_FUTHARK )
    find_dependency( Futhark )
 endif()
-find_dependency( Acts )
+if ( TRACCC_BUILD_IO )
+   find_dependency( Acts )
+endif()
```
[the remark this answers is not in the document]
Comment (fredevb) [reply cmake/traccc-config.cmake.in:46]: I have added `set( TRACCC_BUILD_IO       @TRACCC_BUILD_IO@ )`

**Comment by stephenswat** [comment]
Can you please rebase this on main, as the float conversion should be fixed in #658.

**Review by stephenswat — APPROVED** [review]

**Pull request merged**